src/scripts/components/App.js

```diff
@@ -42,17 +43,48 @@ class App extends Component<Props, void> {
     super(props)
 
     this.props.initializeApp()
+
+    this.state = {
+      modalContent: Object,
+      showModal: false
+    }
+
+    this.showModal = this.showModal.bind(this)
+    this.closeModal = this.closeModal.bind(this)
+  }
+
+  showModal (View) {
+    this.setState({
+      modalContent: View,
+      showModal: true
+    })
+  }
+
+  closeModal (view) {
+    this.setState({
+      showModal: false
+    })
   }
 
   render () {
     const { match } = this.props
     return (
       <ThemeProvider theme={paratiiTheme}>
         <MainTemplate>
+          <Modal
+            show={this.state.showModal}
+            closeModal={this.closeModal}
+            content={this.state.modalContent}
+          />
+
           <MainHeader />
           <Main>
             <Switch>
-              <Route exact path="/" component={Home} />
+              <Route
+                exact
+                path="/"
+                render={props => <Home showModal={this.showModal} />}
```

Comment:
If I understand correctly, does this mean when we want a component to show a modal we need to explicitly pass a `showModal` prop to it?

I think if we put what modal content we are showing in `redux`, we could avoid this. We could have a piece of state called `modal`, and have actions that would update that piece of state (either to set modal content or clear it). Then, any `connect`-ed component could dispatch one of those actions w/o us having to explicitly pass in a `showModal` prop. The `<Modal />` component would then also be `connect`-ed to `redux` and whenever the modal state indicated that it had modal content to show, it would render it. 

Thoughts on this?

Reply:
Were these changes made/will they be?